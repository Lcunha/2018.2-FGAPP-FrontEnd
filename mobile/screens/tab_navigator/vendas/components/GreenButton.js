import styles from '../styles';
import React from 'react';
import {Text, TouchableOpacity} from 'react-native';


class GreenButton extends React.Component{
  render(){
    return(
      <TouchableOpacity style={styles.customBtnBG}
        onPress={() => this.props.onPress()}>
        <Text style={styles.customBtnText}> {this.props.text} </Text>
      </TouchableOpacity>
    );
  }
}

export default GreenButton;
