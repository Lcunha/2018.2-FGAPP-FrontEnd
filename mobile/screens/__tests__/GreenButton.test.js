import React from 'react';
import GreenButton from '../tab_navigator/vendas/components/GreenButton';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<GreenButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing onPress', () => {
    const onPressMock = jest.fn();
    const wrapper = shallow(<GreenButton onPress = {onPressMock}/>);
    const button = wrapper.find('Styled(Button)').at(0);
    button.simulate('press');
    expect(onPressMock).toBeCalled();
  })
