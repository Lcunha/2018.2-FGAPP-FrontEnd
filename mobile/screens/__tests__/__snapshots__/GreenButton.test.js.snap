// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#0EAC6F",
      "flex": 0,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 100,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "300",
      }
    }
  >
     
     
  </Text>
</View>
`;
